### DISCUSSION TITLE
Profile id is missing when using profile() with Pipedrive provider

### DISCUSSION THREAD

**Discussion by darkcount2011** [nextauthjs/next-auth#7593] [Help] [answered]
Hey everyone,

I am currently trying to setup a pipedrive integration using auth.js' pipedrive provider: https://authjs.dev/reference/core/providers_pipedrive

when I use the following code :
`export const handle = sequence(
	SvelteKitAuth({
		providers: [
				PipedriveProvider({
				clientId: PIPEDRIVE_CLIENT_ID,
				clientSecret: PIPEDRIVE_CLIENT_SECRET,
				profile(profile) {
					return { role: 'pipedriveSalesPerson', ...profile };
				}
			})
		],
		// adapters etc...
		}),`
		
		
imports : `
import PipedriveProvider from '@auth/core/providers/pipedrive';
import { SvelteKitAuth } from '@auth/sveltekit';
`

The error I get is the following:
`
[auth][error][OAuthProfileParseError]: Read more at https://errors.authjs.dev#oauthprofileparseerror
[auth][cause]: TypeError: Profile id is missing in Pipedrive OAuth profile response
at getProfile (... node_modules/@auth/sveltekit/node_modules/@auth/core/lib/oauth/callback.js:110:19)
[auth][details]: {}`

Anyone got an idea why using the profile function causes problems? This is basically the default configuration.

**Comment by darkcount2011** [comment, marked as answer]
Seems like the default configuration is expecting the id to be on the top level of the object returned from Pipedrive. But it is actually in the data property.
returning the id as profile.data.id fixes this error.
PipedriveProvider({
				clientId: PIPEDRIVE_CLIENT_ID,
				clientSecret: PIPEDRIVE_CLIENT_SECRET,
				profile(profile) {
					console.log(profile);
					return { id: profile.data.id, }

**Reply by is-it-ayush** [reply]
Thanks! I ran into the same issue and this helped. I did a `JSON.stringify(profile, null, 2)` and indeed the properties were in the "data" key instead of being on the top level. Rewriting it like this solved the issue,
```ts
      profile(profile) {
        return {
          id: profile.data.id,
          name: profile.data.name,
          username: profile.data.username,
          email: profile.data.email,
          image: profile.data.profile_image_url,
        };
      },
```
